runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoOverlay.java

```diff
@@ -143,21 +145,38 @@ public Dimension render(Graphics2D graphics)
 			.build());
 
 		// Health bar
-		if (lastRatio >= 0)
+		if (lastRatio >= 0 && lastHealthScale > 0)
 		{
 			final ProgressBarComponent progressBarComponent = new ProgressBarComponent();
 			progressBarComponent.setBackgroundColor(HP_RED);
 			progressBarComponent.setForegroundColor(HP_GREEN);
 
 			if (lastMaxHealth != null)
 			{
+				int health = 0;
+				if (lastRatio > 0)
+				{
+					// This is the reverse of the calculation of healthRatio done by the server
+					// which is: healthRatio = 1 + (healthScale - 1) * health / maxHealth (if health > 0, 0 otherwise)
+					// It's able to recover the exact health if maxHealth <= healthScale.
+					int minHealth = (lastMaxHealth * (lastRatio - 1) - 1) / (lastHealthScale - 1) + 1;
+					int maxHealth = (lastMaxHealth * lastRatio - 1) / (lastHealthScale - 1);
```

Comment:
This looks like it would divide by zero

Reply:
Hmm ye. That would be a 1 pixel wide health bar tho, but i can add a check for it.